### DISCUSSION TITLE
MSEGUI vs FPGUI?

### DISCUSSION THREAD

**Discussion by ghost** [fredvs/ideU#63] [Q&A] [answered]
What is the different between them? Which of them is more full featured and well developed? Thank you.

**Comment by fredvs** [comment, marked as answer]
Hello and welcome to ideU!!

MSEgui and fpGUI are widgetsets written in Pascal that use low GUI system (xlib for Unix and gdi32 for Windows).

Both do not need any dependencies.

fpGUI is very easy to use and needs only one file source for each form.
MSEgui is very complete, bug-free and totally customizable but needs some effort to understand how it works.

I like both, fpGUI is nice for simple applications and perfect for beginners.
MSEgui is the must for complicated GUI  applications, has dockable forms and is very stable.
MSEgui has more than 9500 commits and is deeply used-tested.
